indexing-service/src/main/java/org/apache/druid/indexing/scheduledbatch/ScheduledBatchTaskManager.java

```diff
@@ -0,0 +1,342 @@
+import org.apache.druid.indexer.TaskLocation;
+import org.apache.druid.indexer.TaskStatus;
+import org.apache.druid.indexing.overlord.TaskMaster;
+import org.apache.druid.indexing.overlord.TaskRunner;
+import org.apache.druid.indexing.overlord.TaskRunnerListener;
+import org.apache.druid.java.util.common.DateTimes;
+import org.apache.druid.java.util.common.concurrent.Execs;
+import org.apache.druid.java.util.common.concurrent.ScheduledExecutorFactory;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.EmittingLogger;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.query.DruidMetrics;
+import org.apache.druid.query.http.ClientSqlQuery;
+import org.apache.druid.query.http.SqlTaskStatus;
+import org.joda.time.DateTime;
+import org.joda.time.Duration;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages the task execution and tracking for the scheduled batch supervisor.
+ * <p>
+ * It submits MSQ tasks to the Broker and tracks their execution status. It listens for task state changes and updates
+ * the {@link ScheduledBatchStatusTracker} accordingly.
+ * </p>
+ * <p>
+ * The manager maintains a mapping of scheduled batch supervisors and their associated task scheduler,
+ * so each supervisor can schedule its tasks based on the {@link CronSchedulerConfig}.
+ * It uses a shared single-threaded {@link ScheduledExecutorService} to schedule tasks across all batch supervisors.
+ * </p>
+ * <p>
+ * Note that all task state tracking by the batch supervisor is currently maintained in memory
+ * and is not persisted in the metadata store.
+ * </p>
+ */
+public class ScheduledBatchTaskManager
+{
+  private static final Logger log = new EmittingLogger(ScheduledBatchTaskManager.class);
+
+  private final TaskRunnerListener taskRunnerListener;
+  private final TaskMaster taskMaster;
+  private final ScheduledBatchStatusTracker statusTracker;
+  private final BrokerClient brokerClient;
+  private final ServiceEmitter emitter;
+
+  private final ScheduledExecutorService tasksExecutor;
+
+  private final ConcurrentHashMap<String, ScheduledBatchTask> supervisorToTaskScheduler = new ConcurrentHashMap<>();
+
+  @Inject
+  public ScheduledBatchTaskManager(
+      final TaskMaster taskMaster,
+      final ScheduledExecutorFactory executorFactory,
+      final BrokerClient brokerClient,
+      final ServiceEmitter emitter,
+      final ScheduledBatchStatusTracker statusTracker
+  )
+  {
+    this.taskMaster = taskMaster;
+    this.brokerClient = brokerClient;
+    this.emitter = emitter;
+    this.statusTracker = statusTracker;
+
+    this.tasksExecutor = executorFactory.create(1, "ScheduledBatchTasksExecutor-%s");
```

Comment:
Maybe use the name of this class itself for this executor `ScheduledBatchTaskManager` so that it's easier to interpret the thread names.